src/main/java/seedu/address/model/Model.java

```diff
@@ -92,4 +96,43 @@
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredContactList(Predicate<Contact> predicate);
+
+    //=========== Module ================================================================================
+
+    /**
+     * Returns true if a module with the same identity as {@code module} exists in the TrackIter.
+     */
+    boolean hasModule(Module module);
+
+    /**
+     * Deletes the given module.
+     * The module must exist in the TrackIter.
+     */
+    void deleteModule(Module target);
+
+    /**
+     * Adds the given contact.
+     * {@code module} must not already exist in the TrackIter.
+     */
+    void addModule(Module module);
+
+    /**
+     * Replaces the given module {@code target} with {@code editedModule}.
+     * {@code target} must exist in the TrackIter.
+     * The module identity of {@code editedModule} must not
+     * be the same as another existing module in the TrackIter.
+     */
+    void setModule(Module target, Module editedModule);
+
+    /**
+     * Returns an unmodifiable view of the filtered module list
+     */
+    ObservableList<Module> getFilteredModuleList();
+
+    /**
+     * Updates the filter of the filtered module list to filter by the given {@code predicate}.
+     *
+     * @throws NullPointerException if {@code predicate} is null.
+     */
+    void updateFilteredModuleList(Predicate<Module> predicate);
```

Comment:
Maybe we should add `throws NullPointerException` in the method declaration? Or remove that line in comments.

Reply:
Okay lemme fix it now